### DISCUSSION TITLE
How to disable justify for content in all tables

### DISCUSSION THREAD

**Discussion by cschreib-ibex** [typst/typst#2192] [Q&A] [answered]
I'd like to set `#set par(justify: true)` for the main text in my document, but keep `justify: false` in tables without having to manually format every cell. I haven't been able to find a way to do this. I'm using `tablex`. Anyone has a tip?

**Comment by PgBiel** [comment, marked as answer]
Does something like

```typ
#let tablex(..args) = {
  set par(justify: false)
  tablex(..args)
}
```

at the top of your document (after the tablex import) work?

**Reply by cschreib-ibex** [reply]
Ah but of course... I was trying to access each cell somehow and change the properties there. I was hoping I could make it work that way, so that I could then extend it to apply a global format to the header rows (right now I'm doing it manually). That shielded me from the obvious answer :) Thanks!

NB I had to write it that way to avoid an infinite recursion
```typst
#let actual_tablex = tablex
#let tablex(..args) = {
    set par(justify: false)
    actual_tablex(..args)
}
```

**Reply by PgBiel** [reply]
Nice catch, I didn't consider the possible recursion haha. No problem!

**Comment by johnmapeson** [comment]
Related: https://github.com/typst/typst/issues/2951
